core/src/main/java/org/infinispan/interceptors/CacheLoaderInterceptor.java

```diff
@@ -93,46 +87,22 @@ protected void injectDependencies(PersistenceManager clm, EntryFactory entryFact
 
    @Override
    public Object visitApplyDeltaCommand(InvocationContext ctx, ApplyDeltaCommand command) throws Throwable {
-      if (enabled) {
-         Object key;
-         if ((key = command.getKey()) != null) {
-            loadIfNeeded(ctx, key, command);
-         }
-      }
-      return invokeNextInterceptor(ctx, command);
+      return visitDataCommand(ctx, command);
    }
 
    @Override
    public Object visitPutKeyValueCommand(InvocationContext ctx, PutKeyValueCommand command) throws Throwable {
-      if (enabled) {
-         Object key;
-         if ((key = command.getKey()) != null) {
-            loadIfNeeded(ctx, key, command);
```

Comment:
Actually looking closer it seems then that loadIfNeededAndUpdateStats should be merged into loadIfNeeded so we won't have this issue in the future.  This would prevent ActivationInterceptor visitPutMapCommand from missing a count as well then.